packages/history/index.ts

```diff
@@ -0,0 +1,373 @@
+////////////////////////////////////////////////////////////////////////////////
+//#region TYPES
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Actions represent the type of change to a location value.
+ *
+ * @see https://github.com/remix-run/history/tree/main/docs/api-reference.md#action
+ */
+export enum Action {
+  /**
+   * A POP indicates a change to an arbitrary index in the history stack, such
+   * as a back or forward navigation. It does not describe the direction of the
+   * navigation, only that the current index changed.
+   *
+   * Note: This is the default action for newly created history objects.
+   */
+  Pop = "POP",
+
+  /**
+   * A PUSH indicates a new entry being added to the history stack, such as when
+   * a link is clicked and a new page loads. When this happens, all subsequent
+   * entries in the stack are lost.
+   */
+  Push = "PUSH",
+
+  /**
+   * A REPLACE indicates the entry at the current index in the history stack
+   * being replaced by a new one.
+   */
+  Replace = "REPLACE",
+}
+
+/**
+ * A URL pathname, beginning with a /.
+ *
+ * @see https://github.com/remix-run/history/tree/main/docs/api-reference.md#location.pathname
+ */
+export type Pathname = string;
+
+/**
+ * A URL search string, beginning with a ?.
+ *
+ * @see https://github.com/remix-run/history/tree/main/docs/api-reference.md#location.search
+ */
+export type Search = string;
+
+/**
+ * A URL fragment identifier, beginning with a #.
+ *
+ * @see https://github.com/remix-run/history/tree/main/docs/api-reference.md#location.hash
+ */
+export type Hash = string;
+
+/**
+ * An object that is used to associate some arbitrary data with a location, but
+ * that does not appear in the URL path.
+ *
+ * @deprecated
+ * @see https://github.com/remix-run/history/tree/main/docs/api-reference.md#location.state
+ */
+export type State = unknown;
```

Comment:
can we switch this to `any`. When it's `unknown` it's an enormous pain to work with. With `any` you can just go ahead and use it, or go nuts with type guards if you want more safety on it instead of `unknown` that forces it.

Reply:
I believe that's what I had originally. `unknown` was, uh, unknown to me when I first wrote history in TS. 👍